### PR TITLE
FEAT-#7187: Change "master" branch to "main"

### LINKED ISSUE #7187: Change "master" branch to "main"
We should update all refs to the new default branch.

### PR DESCRIPTION
**Pull request by YarShev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7187  
- [x] tests passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/getting_started/faq.rst [hunk 1]
```diff
@@ -184,7 +184,7 @@ How can I contribute to Modin?
 If you are interested in contributing please check out the :doc:`Contributing Guide</development/contributing>`
 and then refer to the :doc:`Development Documentation</development/index>`,
 where you can find system architecture, internal implementation details, and other useful information.
-Also check out the `Github`_ to view open issues and make contributions.
+Also check out the `Github`_ to view open issues and make contributionss.
```
Comment (YarShev) [inline docs/getting_started/faq.rst:187]:
```suggestion
Also check out the `Github`_ to view open issues and make contributions.
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -234,10 +234,10 @@ jobs:
         run: |
           git remote add upstream https://github.com/modin-project/modin.git
           git fetch upstream
-          if git diff upstream/master --name-only | grep -q "^asv_bench/"; then
+          if git diff upstream/main --name-only | grep -q "^asv_bench/"; then
               # ASV correctly creates environments for testing only from the branch
-              # with `master` name
-              git checkout -b master
+              # with `main` name
+              git checkout -b main
```
Comment (anmyachev) [inline .github/workflows/ci.yml:240]: Will be changed in https://github.com/modin-project/modin/pull/7190

File: docs/conf.py [hunk 3]
```diff
@@ -122,8 +122,8 @@ def noop_decorator(*args, **kwargs):
 # source_suffix = ['.rst', '.md']
 source_suffix = ".rst"
 
-# The master toctree document.
-master_doc = "index"
+# The main toctree document.
+main_doc = "index"
```
Comment (anmyachev) [inline docs/conf.py:126]: Are you sure?

Comment (YarShev) [reply docs/conf.py:126]: This should stay master_doc. Thanks!

**Review by anmyachev — APPROVED** [review]

**Pull request merged**